pkg/basetypes.go

```diff
@@ -154,3 +154,25 @@ type ReportedUpkeep struct {
 }
 
 type BlockHistory []BlockKey
+
+type UpkeepState uint8
+
+const Performed UpkeepState = iota
+
+type UpkeepStateReader interface {
+	// SelectByID retrieves a single upkeep state
+	SelectByID(ID string) (*UpkeepPayload, UpkeepState, error)
+	// SelectByBlock retrieves upkeep states at a specific block
+	SelectByBlock(block int64) ([]*UpkeepPayload, []UpkeepState, error)
+	// SelectByBlockRange retrieves upkeep states within block range from start (inclusive) to end (exclusive)
+	SelectByBlockRange(start, end int64) ([]*UpkeepPayload, []UpkeepState, error)
+	// SelectByUpkeepID retrieves upkeep states for an upkeep
+	SelectByUpkeepID(upkeepId *big.Int) ([]*UpkeepPayload, []UpkeepState, error)
+	// SelectByUpkeepIDs retrieves upkeep states for provided upkeeps
+	SelectByUpkeepIDs([]*big.Int) ([]*UpkeepPayload, []UpkeepState, error)
+}
+
+type UpkeepStateUpdater interface {
+	SetUpkeepState(UpkeepPayload, UpkeepState) error
+	SetUpkeepStates([]UpkeepPayload, []UpkeepState) error
+}
```

Comment:
The pattern for interfaces we are following is to put the interface in the same file as the function that requires it as a dependency. The main consumer of this interface will be in `pkg/v3/plugin/delegate.go` in the config.

Also, this interface should be reduced to only `SetUpkeepState` for now as that is the only function the plugin will need to start with. The implementation of this interface can still include the other if desired, but let's keep the interface consumed by the plugin simple for now.

Reply:
done